Add documentation for THM kernels and scalar kernels (#25077)
idaholab/moose: modules/thermal_hydraulics/doc/content/source/kernels/ADHeatConductionRZ.md

# ADHeatConductionRZ

!syntax description /Kernels/ADHeatConductionRZ

The equation term added is the same as for the [HeatConduction.md] kernel,

> except that the residual and Jacobian contributions are derived for cylindrical coordinates, the axis and origin

Author: ```suggestion
except that the residual and Jacobian contributions are integrated with cylindrical coordinates, the axis and origin
```
prefer that?

Reviewer: Yeah
